wordpress-mobile/WordPress-Android
# Binary deps and a flag for gutenberg sources

This PR tweaks the build configs so gutenberg-mobile and its dependencies are included as binary dependencies. A gradle property is also introduced as a "switch" to using local, source-level dependencies.

Sibling PRs:
https://github.com/wordpress-mobile/react-native-aztec/pull/80
https://github.com/wordpress-mobile/react-native-recyclerview-list/pull/12
https://github.com/wordpress-mobile/gutenberg-mobile/pull/284


Changes in this PR:

1. Travis config that avoids pulling any git submodules (which does by default), no API 26 tools/platform needed and no need for Node either

2. No dependency on RN or friends anymore. The only dependency is the `gutenberg-mobile` one.

3. EditPostActivity doesn't have a RN dependency on `DefaultHardwareBackBtnHandler` anymore

4. A `gradle.properties` flag is introduced `wp.BUILD_GUTENBERG_FROM_SOURCE` that switches the gradle configs to use source level dependencies and projects. The git submodules need to be updated and yarn install should also be up-to-date so all projects are found.

5. Using a newly forked repo of the `react-native-svg` project to add support for building the lib via jitpack without the need to install RN "locally". Fork lives at https://github.com/wordpress-mobile/react-native-svg and the branch used is [this](https://github.com/wordpress-mobile/react-native-svg/commits/feature/enable-android-build-of-standalone-package).


## To test:

### Test A

0. Make sure `gradle.properties` doesn't define the `wp.BUILD_GUTENBERG_FROM_SOURCE` flag
1. Build and run wpandroid in any flavor and it should run fine. It should be able to open the (block) editor on Gutenberg posts and offer editing as normal.

### Test B

1. Add the `wp.BUILD_GUTENBERG_FROM_SOURCE` flag in `gradle.properties` and set it to `true`
2. Run `git submodule update --init --recursive` to pull in all the gutenberg-mobule submodules
3. Change into `libs/gutenberg-mobile` and run `yarn install` to install the NPM packages
4. Run `yarn wpandroid` to build and launch wpandroid. It should be able to open the (block) editor on Gutenberg posts and offer editing as normal.
5. Notice that the RN app was fetched from the (automatically spawned) Metro server running in its own terminal window.
